mask/eolearn/mask/masking.py

```diff
@@ -11,24 +11,32 @@
 This source code is licensed under the MIT license found in the LICENSE
 file in the root directory of this source tree.
 """
-from typing import Callable, Union
+from __future__ import annotations
+
+from typing import Callable, List, Union
 
 import numpy as np
 
-from eolearn.core import EOTask, ZipFeatureTask
+from eolearn.core import EOPatch, EOTask, FeatureType, ZipFeatureTask
+from eolearn.core.types import FeatureSpec, FeaturesSpecification, SingleFeatureSpec
 
 
 class JoinMasksTask(ZipFeatureTask):
     """Joins together masks with the provided logical operation."""
 
-    def __init__(self, input_features, output_feature, join_operation: Union[str, Callable] = "and"):
+    def __init__(
+        self,
+        input_features: FeaturesSpecification,
+        output_feature: FeatureSpec,
```

Comment:
I noticed that `ZipFeatureTask` uses the `FeaturesSpecification`, but it should be `FeatureSpec`. Please fix the core tasks while we're at it.

Reply:
Also I'd say that if it goes through a parser, `SingleFeatureSpec` is the type to use :thinking:

Reply:
updated